test/mitmproxy/net/http/test_headers.py

```diff
@@ -88,6 +88,12 @@ def test_replace_with_count(self):
         headers = Headers(Host="foobarfoo.com", Accept="foo/bar")
         replacements = headers.replace("foo", "bar", count=1)
         assert replacements == 1
+        assert headers["Host"] == "barbarfoo.com"
+        assert headers["Accept"] == "foo/bar"
+        replacements = headers.replace("foo", "bar", count=2)
+        assert replacements == 2
+        assert headers["Host"] == "barbarbar.com"
+        assert headers["Accept"] == "bar/bar"
```

Comment:
This test does not really verify that we're doing the correct thing now, is it? It seems to pass with and without the patch.

Reply:
* Maybe because the test is committed ahead of my patch? should I open a new PR for test? or commit the test first to verify the issue #3833 and then commit the patch for the issue?
* line 91 and 92 actually make sense to the issue #3833. is the line 93 to 96 unnecessary?

Reply:
Sorry, I got a bit confused here. In short:

1. Yes, this looks like we already had a test for this, but this test checked the wrong things (it just checked the replacement count). You fixed that correctly with lines 91/92.
2. 93-96 are just duplicating what is done before. That somehow confused me, but that's my fault. :) I'll take them out and we're good here.